samples/echobot/doc.go

```diff
@@ -0,0 +1,40 @@
+/*Bot Framework echo bot sample.
+This bot uses msbotbuilder-go: https://github.com/infracloudio/msbotbuilder-go. It shows
+how to create a simple bot that accepts input from the user and echoes it back.
+
+
+Run the example
+
+Bring up a terminal and run. Set two variables for the session as APP_ID and APP_PASSWORD to the values of
+your BotFramework app_id and password. Then, run:
+		go run main.go
+This will start a server which will listen on port 3978
+
+
+Understanding the example
+
+The program starts by creating a hanlder struct of type `activity.HandlerFuncs`.
+This struct contains defination for the `OnMessageFunc` field which is a treated as a callback by the library
+on the respective event.
+	var customHandler = activity.HandlerFuncs{
+	main: func(turn *activity.TurnContext) (schema.Activity, error) {
+			activity := turn.Activity
+			activity.Text = "Echo: " + activity.Text
+			return turn.TextMessage(activity), nil
+		},
+	}
+
+The `init` function picks up the APP_ID and APP_PASSWORD methods from the environment session
+and creates an `adapter` using this.
+
+A webserver is started with a hanlder passed the received payload to `adapter.ParseRequest`
+This methods authenticates the payload, parses the request and returns an Activity value.
+	activity, err := adapter.ParseRequest(ctx, req)
+
+The Activity is then passed to `adapter.ProcessActivity` with the hanlder created to process
+the activity as per the hanlder functions and send the response to the connector service.
+	err = adapter.ProcessActivity(ctx, activity, customHandler)
+
+In case of no error, this web responds with a 200 status
+*/
+package main
```

Comment:
@sahil-lakhwani We can create a README in this repo and put the documentation there.

Reply:
Hey, @PrasadG193 Yes we can do that, I chose this so that generated go doc will have the details in it's format.

Reply:
Let's keep both.

Reply:
Also add steps for starting `ngrok`.

Reply:
Added a README as part of this branch https://github.com/infracloudio/msbotbuilder-go/tree/readme